activerecord/test/cases/adapters/sqlite3/sqlite3_adapter_test.rb

```diff
@@ -245,7 +245,7 @@ def test_insert_logged
           name = "foo"
 
           sqlite_version_query = ["SELECT sqlite_version(*)", "SCHEMA", []]
-          pragma_query = ["PRAGMA table_info(\"ex\")", "SCHEMA", []]
+          pragma_query = ["PRAGMA table_xinfo(\"ex\")", "SCHEMA", []]
```

Comment:
Does SQLite version 3.30.1 or an older version that does not support generated columns have the `table_xinfo` pragma?

Reply:
`PRAGMA table_xinfo` was added in version [3.26.0 (2018-12-01)](https://www.sqlite.org/changes.html#version_3_26_0). So, if the database supports generated columns (added in version [3.31.0 (2020-01-22)](https://www.sqlite.org/changes.html#version_3_31_0)), then it supports `table_xinfo`.